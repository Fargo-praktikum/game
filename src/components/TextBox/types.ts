import { InputHTMLAttributes } from "react";

export type TTextBoxProps = InputHTMLAttributes<HTMLInputElement>;
