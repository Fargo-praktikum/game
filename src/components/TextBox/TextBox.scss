.text-box {
    border: 0;
    box-sizing: border-box;
    width: 100%;

    &:focus {
        outline: 0;
    }
}
