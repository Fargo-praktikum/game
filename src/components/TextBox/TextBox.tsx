import React from "react";
import { TTextBoxProps } from "./types";

import "./TextBox.scss";

export const TextBox = (props: TTextBoxProps): JSX.Element => {

    const cssClasses = React.useMemo(() => {
        return `text-box ${props.className ?? ""}`;
    }, [props.className]);

    return (
        <input
            {...props}
            className={cssClasses}
        />
    );
};
