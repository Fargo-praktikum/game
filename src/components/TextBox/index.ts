import { TextBox } from "./TextBox";

export { TextBox };
