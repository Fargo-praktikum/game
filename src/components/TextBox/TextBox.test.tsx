import React from "react";
import renderer from "react-test-renderer";

import { TextBox } from "./TextBox";

describe("Проверка TextBox", () => {
    it("корректная отрисовка", () => {
        const textBox = renderer
            .create(<TextBox/>);
        expect(textBox.toJSON()).toMatchSnapshot();
    });

    it("отрисовка с пропсами", () => {
        const field = {
            value1: "test",
            name: "testName",
            multiple: true,
            checked: false,
            onChange: () => { return; },
            onBlur: () => { return; },
        };
        const textBox = renderer
            .create(
                <TextBox
                    type={"text"}
                    className="form__textinput-profile right"
                    {...field}
                />
            );
        expect(textBox.toJSON()).toMatchSnapshot();
    });

});



