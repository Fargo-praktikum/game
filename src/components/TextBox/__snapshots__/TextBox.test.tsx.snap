// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Проверка TextBox корректная отрисовка 1`] = `
<input
  className="text-box "
/>
`;

exports[`Проверка TextBox отрисовка с пропсами 1`] = `
<input
  checked={false}
  className="text-box form__textinput-profile right"
  multiple={true}
  name="testName"
  onBlur={[Function]}
  onChange={[Function]}
  type="text"
  value="test"
/>
`;
